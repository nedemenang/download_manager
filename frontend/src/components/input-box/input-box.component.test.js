import React from "react";
import { Provider } from "react-redux";
import InputBox from "./input-box.component";
import { shallow, configure, mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

configure({ adapter: new Adapter() });

const store = mockStore({
  downloads: [],
});

let enzymeWrapper = shallow(
  <Provider store={store}>
    <InputBox />
  </Provider>
);

describe("<InputBox />", () => {
  beforeEach(() => {
    enzymeWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <InputBox />
        </BrowserRouter>
      </Provider>
    );
  });

  test("should mount", () => {
    mount(
      <Provider store={store}>
        <BrowserRouter>
          <InputBox />
        </BrowserRouter>
      </Provider>
    );
  });

  test("should matches the stored snapshot", () => {
    expect(enzymeWrapper.text()).toMatchSnapshot();
  });

  test("should render 1 input tags", () => {
    expect(enzymeWrapper.find("input")).toHaveLength(1);
  });

  test("should render 1 link with class name fas", () => {
    expect(enzymeWrapper.find(".searchbtn")).toHaveLength(1);
  });
});
