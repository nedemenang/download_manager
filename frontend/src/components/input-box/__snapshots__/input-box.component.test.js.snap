// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputBox /> should matches the stored snapshot 1`] = `""`;
